lib/bolt/transport/ssh.rb

```diff
@@ -16,18 +16,21 @@ def initialize
         rescue LoadError
           logger.debug("Authentication method 'gssapi-with-mic' (Kerberos) is not available.")
         end
-
-        @transport_logger = Logging.logger[Net::SSH]
-        @transport_logger.level = :warn
       end
 
       def with_connection(target)
-        conn = Connection.new(target, @transport_logger)
+        conn = if target.transport_config['ssh-command']
+                 ExecConnection.new(target)
+               else
+                 transport_logger = Logging.logger[Net::SSH]
+                 transport_logger.level = :warn
+                 Connection.new(target, transport_logger)
+               end
         conn.connect
         yield conn
       ensure
         begin
-          conn&.disconnect
+          conn&.disconnect if conn.is_a?(Connection)
```

Comment:
Is that necessary before we have control sockets? This conditional is a bit weird, but I'm concerned about defining an api for transports that doesn't always make sense. A disconnect method doesn't seem generally necessary for implementations of the SSH connection in the same way the connect method is.

Reply:
I generally prefer to keep the differences encapsulated in the two connection classes rather than bubbling up to this level.

The `connect` method is a bit weird here, in that it doesn't actually _start_ a connection, it just tests whether a connection can be made. But the use of `with_connection` to implement `connected?` is part of the `Bolt::Transport::Simple` base class so it's not as obvious to change that.

The only consumer of the connections is this class itself, so I'm not really concerned about defining an API one way or the other. We can freely change it as it evolves.

Reply:
That's true, I doubt this will set a precedent in stone if we ever make the API open, which is a huge if anyway. And if we did, it's easy to change. Updated!